lib/internal/errors.js

```diff
@@ -828,16 +765,21 @@ function setArrowMessage(err, arrowMessage) {
 // Hide stack lines before the first user code line.
 function hideInternalStackFrames(error) {
   overrideStackTrace.set(error, (error, stackFrames) => {
-    let frames = stackFrames;
+    let result = '';
     if (typeof stackFrames === 'object') {
-      frames = ArrayPrototypeFilter(
-        stackFrames,
-        (frm) => !StringPrototypeStartsWith(frm.getFileName() || '',
-                                            'node:internal')
-      );
+      for (const frame of stackFrames) {
+        const filename = frame.getFileName();
+        if (!filename || !StringPrototypeStartsWith(filename, 'node:internal')) {
+          result += `\n    at ${frame}`;
+        }
+      }
+    } else {
+      for (const frame of stackFrames) {
+        result += `\n    at ${frame}`;
+      }
     }
-    ArrayPrototypeUnshift(frames, error);
-    return ArrayPrototypeJoin(frames, '\n    at ');
+    result = error + result;
+    return result;
```

Comment:
```suggestion
    return error + result;
```
assuming this is either the same or faster?

Reply:
It's identical.

Reply:
consider this a nit then to minimize reassignment :-)

Reply:
Done